src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -239,6 +239,62 @@ impl BDA {
     }
 }
 
+fn ok_err_static_header_handling<F>(
+    loc_1: Option<StaticHeader>,
+    loc_2: StratisError,
+    location: MetadataLocation,
+    f: &mut F,
+) -> StratisResult<Option<StaticHeader>>
+where
+    F: Read + Seek + SyncAll,
+{
+    if let Some(loc_1) = loc_1 {
+        loc_1.write(f, location)?;
+        Ok(Some(loc_1))
+    } else {
+        // Location 1 doesn't have a signature, but location 2 did, but it got an error,
```

Comment:
Please drop the inline comment and put a header comment instead, something like:

"""
Action taken when one sigblock is interpreted as invalid.
If the other sigblock is interpreted as a Stratis header, attempt repair
of the invalid sigblock, returning an error if that fails, otherwise returning
the valid sigblock. In all other cases, return the error associated with
the invalid sigblock.
"""

This is also a little odd, for the same reasons as the other one,
but again, we should leave it alone for now and just complete
the refactoring.